src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -42,45 +38,37 @@ internal sealed class SocketConnectionListener : IConnectionListener
             var maxWriteBufferSize = _options.MaxWriteBufferSize ?? 0;
             var applicationScheduler = options.UnsafePreferInlineScheduling ? PipeScheduler.Inline : PipeScheduler.ThreadPool;
 
+            var logger = loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets");
+            _trace = new SocketsTrace(logger);
+
             if (ioQueueCount > 0)
             {
-                _settingsCount = ioQueueCount;
-                _settings = new Settings[_settingsCount];
+                _factoryCount = ioQueueCount;
+                _factories = new SocketConnectionContextFactory[_factoryCount];
```

Comment:
I think it would be better to have the `SocketConnectionContextFactory` manage its on set of `IOQueues` rather than forcing any implementer of a custom Socket-based `IConnectionListenerFactory` to basically do this themselves or suffer worse perf.

I know @davidfowl said he didn't think we should have `SocketConnectionOptions.IOQueueCount` though which we would need if `SocketConnectionContextFactory` manages its on set of `IOQueues`. Maybe we should set up a meeting with the three of us.